.github/CODE_STANDARDS.md

```diff
@@ -0,0 +1,295 @@
+
+   ```go
+   func TestSomethingVeryUsefulIsHappening(t *testing.T) {
+       //given
+       ... // prepare inputs, mocks, application state
+   
+       //when
+       ... // call the function you're actually testing
+   
+       //then
+       ... // check expectations about the function output
+   }
+    ```
+
+4. **Test Isolation**: Ensure test isolation by avoiding the use of global variables and shared state. Each test should be independent and not rely on the execution of other tests. If a test requires a shared state, use a setup function to create the state before each test.
+
+5. **Test Data**: Avoid random data in tests. Instead, use predefined data to ensure test consistency and reproducibility. If random data is required, use a seed to ensure the same data is generated each time the test is run.
+
+6. **Test Cases**: If you are writing a public functions - it should be covered by tests. We should have test cases for all possible scenarios. That means that **we should have tests for all possible errors** that can be returned from the function.
+Of course not only error paths should be covered - **we should highlight the happy path as well**.
+
+7. **Testing private (unexported) functions**: When testing private functions, we should test them through the public (exported) functions that use them. The exception is when the private function is too complex to be tested through the public function. Good example is for example a function that is implementing a complex algorithm. In this case we should test the private function directly.
+
+### 2.3 Code Review
+
+#### Guidelines for Code Review
+
+1. **Constructive Feedback**: Reviewers should provide constructive feedback, highlighting both strengths and areas of improvement. Comments should be clear, concise, and related to code structure, functionality, or style.
+
+2. **Coding Standards**: Both authors and reviewers should ensure the code adheres to established coding standards, including formatting, naming conventions, and best practices as outlined in official Golang guidelines.
+
+3. **Testing**: Reviewers should ensure that the submitted code is accompanied by adequate tests, covering both happy paths and error paths. Check the readability, descriptiveness, and structure of tests.
+
+4. **Error Handling**: Pay special attention to error handling within the code. Ensure that errors are not ignored, logged appropriately, and presented to the user in a user-friendly format when applicable.
+
+5. **Performance**: Review the code for any potential performance issues, such as inefficient loops, unnecessary allocations, or misuse of concurrency.
+
+6. **Dependency Management**: Ensure that any new dependencies are necessary, appropriately versioned, and have been vetted for performance and security.
+
+7. **Security Practices**: Scrutinize the code for any security vulnerabilities. Validate input, avoid security anti-patterns, and utilize secure coding practices.
```

Comment:
Is this a clue for the reviewer? seems more like for the dev/author

Reply:
Replaced with point that is targeted to reviewers